### ISSUE TITLE
Nested structs & param expansion

### ISSUE THREAD

**Issue by jacogr** [polkadot-js/apps#1652]
Recently we have catered for recursive structs in the API. On the UI side however, the UI continues to display `Bytes` for these, i.e. the hex input which basically says "I give . up, I don't know what to do with this"

Even with just nested structs, or stucts with enums, there are def. gaps. For recursive types (done via `Box`), we have a node-template sample available.

Example repo -

https://github.com/JoshOrndorff/substrate/tree/picalc

Updated apps repo with the new API (& types) -

https://github.com/JoshOrndorff/apps/tree/api-0.93.0-beta1

**Comment by maxsam4** [comment]
The extrinsic submission form/page also doesn't work properly with nested structs.

**Comment by jacogr** [comment]
Yes, there is a lot of work left - a big issue with these is that as soon as I have an (a) actual customised node with the customized types and (b) a public RPC (so I can actually test), I do try and make sure that problematic calls and structs are actually usable.

... Or if there is a specific one in substrate base where this is problematic.

The testing of this is a big issue, in most cases it is blind "this should be better" changes. Sadly, just cannot afford to have even more random branches/repos locally, not because I'm lazy, but rather there is a limit as to how many is actually feasible and just between Polkadot/Substrate and changes it is, well, close to 10.

TL;DR Going has been really slow on these and will probably continue that way. Recently there . has been some nice improvements made based on external nodes, but there are a lot of gaps :(
